pkg/providers/instancetype/instancetypes.go

```diff
@@ -145,10 +145,14 @@ func instanceTypeZones(sku *skewer.SKU, region string) sets.Set[string] {
 	// skewer returns numerical zones, like "1" (as keys in the map);
 	// prefix each zone with "<region>-", to have them match the labels placed on Node (e.g. "westus2-1")
 	// Note this data comes from LocationInfo, then skewer is used to get the SKU info
+	// If an offering is non-zonal, the availability zones will be empty.
```

Comment:
Very interesting discovery so certain offerings are not zonal and only have one location? Or are they physically hosted in all of the locations you just can't request a specific location? How does this work?

Reply:
> What is not clear to me, is whether non-zonal offering is always available (as long as SKU/size is in the list); would be good to check - especially since the proposed logic makes this assumption. This will also be important longer term, if we decide to expand support for non-zonal offerings alongside zonal, and not just as fallbacks.

This is a good point. I'll try to find some docs/ask around AKS to understand the context here + ensure that this is a fair assumption. I'm going to proceed with merging for now to unblock Heba.